src/dockerfile/structure/dockerfile_parser.go

```diff
@@ -0,0 +1,274 @@
+			tag := generateTag(line)
+			for _, tagItem := range tags {
+				if tagItem.Key == tag.Key {
+					if tagItem.Value != tag.Value && tagItem.Remove == false {
+						newLine := "LABEL " + tagItem.Key + "=" + "\"" + tagItem.Value + "\"\n"
+						newFile = newFile + newLine
+						fileChanged = true
+					}
+					if tagItem.Value == tag.Value && tagItem.Remove == false {
+						newFile = newFile + line + "\n"
+						fileChanged = true
+					}
+					if tagItem.Remove {
+
+					}
+				}
+			}
+		} else {
+			newFile = newFile + line + "\n"
+		}
+	}
+	for _, tagItem := range tags {
+		if !tagItem.Exists {
+			newLine := "LABEL " + tagItem.Key + "=" + "\"" + tagItem.Value + "\"\n"
+			newFile = newFile + newLine
+			fileChanged = true
+		}
+	}
+	if fileChanged {
+
+		var fileError = os.WriteFile(writeFilePath, []byte(newFile), 0644)
+		if fileError != nil {
+			var errorMessage = fmt.Errorf("failed to read file %s because %s", writeFilePath, fileError)
+			log.Fatal(errorMessage)
+		}
+	}
+
+	return nil
+}
+
+func existsInNewLabels(tag TagItem, block structure.IBlock) bool {
+	newLabels := block.GetNewTags()
+	for _, item := range newLabels {
+		if tag.Key == item.GetKey() {
+			return false
+		}
+	}
+	return true
+}
+
+func checkItemValue(tag TagItem, list []TagItem) []TagItem {
+	index := 0
+	for _, item := range list {
+		if item.Key == tag.Key {
+			if item.Value != tag.Value {
+				var updatedTag TagItem
+				updatedTag.Key = tag.Key
+				updatedTag.Value = tag.Value
+				updatedTag.NeedsUpdating = true
+				updatedTag.Exists = true
+				list[index] = updatedTag
+			}
+		}
+		index += 1
+	}
+	return list
+}
+
+func existsInOldLabels(key string, list []TagItem) bool {
+	for _, item := range list {
+		if item.Key == key {
+			return true
+		}
+	}
+	return false
+}
+
+func (p *DockerfileParser) GetSkippedDirs() []string {
+	var ignoredDirs []string
+	return ignoredDirs
+}
+
+func (p *DockerfileParser) Close() {
+	logger.MuteOutputBlock(func() {
+		p.providerToClientMap.Range(func(provider, iClient interface{}) bool {
+			client := iClient.(tfschema.Client)
+			client.Close()
+			return true
+		})
+	})
+}
+
+func (p *DockerfileParser) GetSupportedFileExtensions() []string {
+	return []string{common.DockerFileType.Extension}
+}
+
+func (p *DockerfileParser) Init(rootDir string, args map[string]string) {
+	p.rootDir = rootDir
+	//p.dockerFileExists = false
+	//p.IngestFile()
+	//p.ReadYamlConfiguration()
+	//p.PatchDockerFile()
+
+}
+
+func generateTag(line string) tags.Tag {
+	reg := regexp.MustCompile(`[0-9a-zA-Z\"]+=[0-9a-zA-Z@.\"\s]+`)
+	result := reg.FindString(line)
+	keyValue := strings.Split(result, "=")
+	if len(keyValue) < 2 {
+		log.Fatal("Line in Dockerfile could not be parsed correctly.")
+	}
+	tag := tags.Tag{
+		Key:   FormatString(keyValue[0]),
+		Value: FormatString(keyValue[1]),
+	}
+	return tag
+}
+
+func readFile(filePath string) ([]string, error) {
+	src, err := ioutil.ReadFile(filePath)
```

Comment:
@sshintaku Please merge from master if you haven't, leverage golang 1.19, and then:
```suggestion
	src, err := os.ReadFile(filePath)
```